azurelinuxagent/common/protocol/wire.py

```diff
@@ -768,6 +796,35 @@ def _update_from_goal_state(self, refresh_type):
 
         raise ProtocolError("Exceeded max retry updating goal state")
 
+    def _save_goal_state_if_necessary(self, new_goal_state, refresh_type):
+        # We cache the goal state if one of the following is true
```

Comment:
Let's update this comments. These files are not a cache (we removed the use of the cache a few releases ago). They are used for debugging purposed only.